server/controllers/profile.js

```diff
@@ -0,0 +1,60 @@
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+exports.profile_create_post = asyncHandler(async (req, res, next) => {
+  const { firstName, lastName, description, availability } = req.query;
+
+  if (!firstName || !lastName) {
+    res.status(400);
+    throw new Error("Invalid name");
+  }
+  await Profile.create({
+    firstName,
+    lastName,
+    description,
+    availability,
+  });
+});
+
+exports.profile_update_post = function (req, res) {
+  const { firstName, lastName, description, availability, id } = req.query;
+  const idExists = await Profile.findOne({ id });
+
+  if (idExists) {
+    await Profile.updateOne(
+      { _id: id },
+      {
+        firstName,
+        lastName,
+        description,
+        availability,
+      }
+    );
+    res.status(201);
+  } else {
+    res.status(400);
+    throw new Error("Invalid profile update");
+  }
+};
+
+exports.profile_get = asyncHandler(async (req, res, next) => {
+  const { id } = req.query;
+
+  let profile;
```

Comment:
Please consider using 'const' instead of 'let' as I don't see this variable being reassigned.